alvr/server/src/connection.rs

```diff
@@ -671,14 +693,15 @@ fn try_connect(mut client_ips: HashMap<IpAddr, String>) -> ConResult {
                     return;
                 };
 
+                let data_manager_lock = SERVER_DATA_MANAGER.read();
+                let config = &data_manager_lock.settings().headset;
```

Comment:
Do not call this here. `SERVER_DATA_MANAGER.read()` will hold the lock until the end of the loop, but we want to unlock it sooner in case there is a write pending in another thread. if needed, call this multiple times in each `{ }` block (sub scope).